src/target/cortexa.c

```diff
@@ -91,6 +91,19 @@ typedef struct cortexa_priv {
 #define CORTEXAR_DBG_WCR   0x1c0U
 #define CORTEXAR_CTR       0xd04U
 
+#define CORTEXAR_DBG_DSCCR 0x028U
+#define CORTEXAR_DBG_DSMCR 0x02cU
+#define CORTEXAR_DBG_OSLAR 0x300U
+#define CORTEXAR_DBG_OSLSR 0x304U
+#define CORTEXAR_DBG_LAR   0xfb0U /* Lock Access */
+#define CORTEXAR_DBG_LSR   0xfb4U /* Lock Status */
+
+#define DBGOSLSR_OSLM0 (1U << 0U)
+#define DBGOSLSR_OSLK  (1U << 1U)
+#define DBGOSLSR_NTT   (1U << 2U)
+#define DBGOSLSR_OSLM1 (1U << 3U)
+#define DBGOSLSR_OSLM  (DBGOSLSR_OSLM0 | DBGOSLSR_OSLM1)
```

Comment:
These should be prefixed with `CORTEXAR_` and the `_` between `DBG` and the register name preserved, but given how much still needs rewriting in this file, we're going to let that slide if you don't feel like fixing it - this is just a note so it's not forgotten.

Reply:
I understand the item but other bitfields just 20 lines down are written that way, and I complied with that style to not have to change all of it.

Reply:
Those other fields are going to be changed in the rewrite, follow the style set in cortexr.c from the Cortex-R support feature branch please as that is the style this code will be switching to.

Reply:
I can rename the 5 macros in anticipation of code style switch iff it is planned to happen before v1.10 release (and I don't know the plan).
nTT is unused, should I drop that bit?

Reply:
It will happen after the release, but not long after - it would be better if it's done here to reduce churn and preserve attribution as much as possible.

Regarding the nTT bit, please keep it as that should become used when we extend things out to support 64-bit processors.

Reply:
Rename complete.
I don't honestly like the OSLM bitmask and the (know broken into 2 lines by clang-format) check down on L591. Why did they do it that way and can I just use individual bits in the logic condition, dropping the bitmask altogether? OSLM bits [3,0] value 0b11 is reserved anyways.

Reply:
Yeah, we couldn't figure out why that happened or what thinking was involved there either - it just makes everything a bit more difficult around those bits. We can try to refine this in v2.0 as we'd like to get this merged and the main STM32MP15 PR too - that also gives more time to do research and figure out the best approach here.